include/net/wifi_ready.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+
+#ifndef WIFI_READY_H__
+#define WIFI_READY_H__
+
+#define WPA_SUPP_EVENTS (NET_EVENT_WPA_SUPP_READY) | \
+			(NET_EVENT_WPA_SUPP_NOT_READY)
+
+typedef void (*start_app_callback_t)();
```

Comment:
Starting Application or not is application's call, this should just be `wifi_ready_callback` with `bool wifi_ready`. You haven't handled Wi-Fi not ready case.